frontend/frontend-react/src/components/CreateRackForm.js

```diff
@@ -63,48 +28,76 @@ export class CreateRackForm extends Component {
 
-    if ((validNumRegex.test(rack_num))) {
-      axios.post('/api/racks/', stateToSend)
-        .then(function (response) {
-          //   let message = response.data.results;
-          alert('Creation of ' + rack_num + ' was successful.');
-        })
-        .catch(function (error) {
-          alert('Creation was not successful.\n' + JSON.stringify(error.response.data, null, 2));
-        });
-    } else {
-      alert("Rack Numbers must be specified by a Single Letter Followed by Multiple Numbers.");
+    if(start_rack !== null && (end_rack == null || end_rack === '')){
+      if ((validNumRegex.test(start_rack))) {
+        axios.post('/api/racks/', stateToSend)
+          .then(function (response) {
+            //   let message = response.data.results;
+            alert('Creation of ' + start_rack + ' was successful.');
+          })
+          .catch(function (error) {
+            alert('Creation was not successful.\n' + JSON.stringify(error.response.data, null, 2));
+          });
+      } else {
+        // console.log(this.state.rack_number);
+        alert("Rack Numbers must be specified by a Single Letter Followed by Multiple Numbers.");
+      }
+    }
+    else if(start_rack !== null && end_rack !== null){
+      if ((validNumRegex.test(start_rack) && validNumRegex.test(end_rack))) {
+        axios.post('/api/racks/many/', stateToSend)
+          .then(function (response) {
+            let message = response.data.results;
+            alert(response.data.results);
+            this.props.sendShowTable(true);
+          })
+          .catch(function (error) {
+            alert('Creation was not successful.\n' + JSON.stringify(error.response.data, null, 2));
+          });
+      } else {
+        alert("Rack Numbers must be specified by a Single Letter Followed by Multiple Numbers.");
+      }
     }
-    this.props.sendShowTable(true);
   }
 
   render() {
     return (
-      <div>
-        <Button variant="outlined" onClick={() => this.props.sendShowTable(true)}>Back</Button>{' '}
-        <form onSubmit={this.handleSubmit}>
+      <Container maxwidth="s">
+        <Link to='/racks/'><Button variant="outlined">Back</Button>{' '}</Link>
+        <form onSubmit={this.handleSubmit} action='/racks/'>
           <Grid container spacing={1}>
             <Grid item xs={12}>
-              <h1>Create Single Rack</h1>
+              <h1>Create Rack(s)</h1>
             </Grid>
-            <Grid item xs={12}>
-              <TextField label='Rack Number' type="text" fullWidth
-                         onChange={e => this.setState({rack_number: e.target.value})}/>
+            <Grid item xs={3}>
+              <TextField label='Creation Range Start' type="text" fullWidth
+                         onChange={e => this.setState({rack_num_start: e.target.value, rack_number: e.target.value})}/>
+            </Grid>
+            <Grid item xs={3}>
+              <TextField label='Creation Range End (optional)' type="text" fullWidth
```

Comment:
Why is this optional? Does it work so like if i only want to make one rack I just enter the first number?

If so I think that should be clarified in UI somewhere

Reply:
Yes to the only if you want to make one rack you enter the first number. Do you think I should just have a small line that's like enter the first number to make one rack, enter a later rack to create a full range?

Reply:
that would be super helpful!